### PR TITLE
unit test cases for create credential REST Api

### PR DESCRIPTION
**Pull request by neethi-shashidhar**

### CODE REVIEW

**Comment by hp-networking** [comment]
test_bnp_credential.py:
_mechanism_drivers - not required. and line 54 also you can remove . it is unnecessary
also extension_drivers cfg override not required
 self.snmp_req1  & self.snmp_body duplicate contents repeated. I would suggest you form the request wherever it is necessary.it is applicable to netconf too. 
you can manufacture the request in create_credentials_for_snmp function

self.dbval included in setUp method. As you know for each test cases setup will invoked and unnecessarily this functions will get called from the DB

Handle for SNMP V1/V2C also
Handle for Netconf Soap

L 97: self.protoval can you declare some meaninful name?
L98 :  self.protoval1 too

create_credentials_for_snmp(self, req, name, dbval, protoval, body 

dbval, protoval - ? can be db_value and protocol_value .
dbval not required to pass in this method you can mock the DB helper inside the contextlib only

function name _create_credentials_for_snmp & _create_snmp_cred is confusing. 
dont you think _create_snmp_cred will be called internally in bnp_credential if we invoke create request?

above comments applicable to netconf
L170 Reuse the body from earlier declaration

L 187 remove the hardcoded path in /home/sdn/ refer https://github.com/openstack/networking-l2gw/blob/master/networking_l2gw/tests/unit/services/l2gateway/agent/ovsdb/test_manager.py

how about other cases where 
1. invalid protocol
2. by giving any extra keys in body in that case you should expect HTTPBadRequest
3. not giving name
4. invalid snmpv3 parameters - some missing keys or profile
5. invalud netconf parameters - give some missing password
baremetal_network_provisioning/tests/unit/db/test_bm_nw_provision_db.py
L229 :
why status you added. it is removed in the latest code
switch_dict  - why you have modified as per old spec?
L322: status is not exist in DB schema

test_update_bnp_phys_switch_access_params not supported in extension , how it will work?

add the unit test cases for get_bnp_phys_switch_by_name & delete_bnp_phys_switch_by_name

**Comment by hp-networking** [comment]
test_bnp_credential.py
Line 98:
_test_create_credential_for_netconf  & _test_create_credential_for_snmp always return None then what is the need to check for None?
    result_netssh = self._test_create_credential_for_netconf(body_netssh,
 +                                                                 None)
 +        result_netsoap = self._test_create_credential_for_netconf(body_netsoap,
 +                                                                  None)
 +        result_netsoap = self._test_create_credential_for_netconf(body_netsoap,
 +                                                                  None)
same is applicable to 78 to 82
L132: where are you creating the many names to check name already exists?
just wondering below assertion really worked
self.assertRaises(webob.exc.HTTPConflict,
 +                          self._test_create_credential_for_snmp,
 +                          body_snmp, 'CRED1')

**Comment by koteswar16** [comment]
Remove all ^M in the files and run pep8 once.

**Comment by selvakumarnms** [comment]
get_all_bnp_switch_ports from DB helper is missing . how it worked?
how unit test case been passed for that method?

**Comment by koteswar16** [comment]
bnpclient/bnp_client_ext/bnpswitch/_bnp_switchport.py:
L29: correct docstring, it lists all switch ports info
db/bm_nw_provision_db.py:
L159: fun name should be get_all_bnp_switch_port_maps
ml2/extensions/bnp_switchport.py:
L59: put key-values in order 0,1,...6
devstack/README.rst:
L26: change ml2_conf_hp.ini to ml2_conf_hp**e**.ini

make sure you didnot delete old unit testcases which are required.

File: baremetal_network_provisioning/ml2/extensions/bnp_switchport.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        filters = req_dict
+        port_maps = db.get_all_bnp_switch_port_maps(context, **filters)
+        port_list = []
+        for port_map in port_maps:
+            port_dict = {'neutron_port_id': port_map[0],
+                         'switch_port_name': port_map[1],
+                         'lag_id': port_map[2],
+                         'segmentation_id': str(port_map[3]),
+                         'access_type': port_map[4],
+                         'bind_status': str(port_map[5]),
```
Comment (selvakumarnms) [inline baremetal_network_provisioning/ml2/extensions/bnp_switchport.py]: bind_status : check for 0 and 1
if value is 0 then it is success 
else Failure
Right now raw values being send to REST .

File: baremetal_network_provisioning/ml2/extensions/bnp_switchport.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+            _("This operation is not allowed"))
+
+
+class Bnp_switchport(extensions.ExtensionDescriptor):
+
+    """API extension for Baremetal Switch port support."""
+
+    @classmethod
+    def get_name(cls):
+        return "Baremetal Switch Ports"
```
Comment (selvakumarnms) [inline baremetal_network_provisioning/ml2/extensions/bnp_switchport.py]: "Baremetal switch port to neutron port mapping"

File: baremetal_network_provisioning/ml2/extensions/bnp_switchport.py (hunk 1 above)
Comment (neethi-shashidhar) [inline baremetal_network_provisioning/ml2/extensions/bnp_switchport.py]: Done.

File: baremetal_network_provisioning/ml2/extensions/bnp_switchport.py (hunk 2 above)
Comment (neethi-shashidhar) [inline baremetal_network_provisioning/ml2/extensions/bnp_switchport.py]: Done.

**Comment by hp-networking** [comment]
Looks good.

**Pull request merged**